app/jobs/csv_export_job.rb

```diff
@@ -34,28 +34,44 @@ def deploy_csv_export(csv_export)
     filename = csv_export.path_file
     filter = csv_export.filters
 
-    @deploys = Deploy.joins(:stage, :job).where(filter)
+    get_deploys(filter)
     summary = ["-", "Generated At", csv_export.updated_at, "Deploys", @deploys.count.to_s]
     filters_applied = ["-", "Filters", filter.to_json]
 
     CSV.open(filename, 'w+') do |csv|
-      csv << [
-        "Deploy Number", "Project Name", "Deploy Sumary", "Deploy Commit", "Deploy Status", "Deploy Updated",
-        "Deploy Created", "Deployer Name", "Deployer Email", "Buddy Name", "Buddy Email",
-        "Production Flag", "No code deployed"
-      ]
+      csv << Deploy.csv_header
       @deploys.find_each do |deploy|
-        csv << [
-          deploy.id, deploy.project.name, deploy.summary, deploy.commit, deploy.job.status, deploy.updated_at,
-          deploy.start_time, deploy.job.user.name, deploy.job.user.try(:email), deploy.buddy_name, deploy.buddy_email,
-          deploy.stage.production, deploy.stage.no_code_deployed
-        ]
+        csv << deploy.csv_line
       end
       csv << summary
       csv << filters_applied
     end
   end
 
+  def get_deploys(filter)
+    if filter.keys.include?('environments.production')
+      production_value = filter.delete('environments.production')
+      # To match logic of stages.production? True when any deploy_group environment is true or
+      # deploy_groups environment is empty and stages is true
+      production_query = "(A.production = ? OR (A.production IS NULL AND stages.production = ?))"
+      production_query = "NOT " + production_query if production_value == false
```

Comment:
Moving the subquery out to a pre-constructed string and labeling it "StageProd" 

stage.production? returns true if any DeployGroup.Environment.production is true or if no Deploy Groups are associated with the stage, Stage.production.

production = true, The query is returning StageProd.production is true or if StageProd is not present the stages.production is true.

production = false, probably could use the same logic with the false values, but there may be some conditions where NOT could give the undefined values as well.

stage_prod null = no associated environment.production for the stage (can never be null from db level validation)
alt logic is not using not and passing false on instead of true

stage | stage_prod | expected from stage.production? | this logic | alt logic
----- | ------------ | --------------------------------- | --------- | --------
any | true | true | true | true
any | false | false | false | false
true | null | true | true | true
false | null | false | false | false
null | null | false | false | never applicable